### PR TITLE
Fixes simple animal emotes / Refactors emote list

### LINKED ISSUE #44455: Space Carp can't emote anymore [EDIT: maybe all simplemob emotes are bugged]
Issue reported from  Round ID: 110345 (/tg/Station Sybil [ENGLISH] [US-WEST] [100% LAG FREE])

Reporting client version: 512.1453

When an admin spawned me as a carp yesterday (as of writing this), I was able to emote.

I can no longer emote now (as of that round ID). This annoys me since it USED to work yet now it doesn't. I tried hotkeys, me ", and IC tab however none of them worked.

edit: okay apparently space carps, holocarps, and the normal clown simplemob can't emote anymore either.

### PR DESCRIPTION
**Pull request by AnturK**
Fixes #44455

While i was at it i made emote list properly support multiple emotes with the same key (right now it's just scream and gnarl)

### CODE REVIEW

**Comment by ishitbyabullet** [comment]
**Hero Unit deployed!**

File: code/_globalvars/lists/mobs.dm [hunk 1]
```diff
@@ -55,3 +56,17 @@ GLOBAL_LIST_EMPTY(mob_config_movespeed_type_lookup)
...
 		M.update_config_movespeed()
+
+/proc/init_emote_list()
+	. = list()
+	for(var/path in subtypesof(/datum/emote))
+		var/datum/emote/E = new path()
+		if(!.[E.key])
+			.[E.key] = list(E)
+		else
+			.[E.key] |= E
```
Comment (ninjanomnom) [inline code/_globalvars/lists/mobs.dm]: You shouldn't need |= here
```suggestion
			.[E.key] += E
```

File: code/_globalvars/lists/mobs.dm [hunk 2]
```diff
@@ -55,3 +56,17 @@ GLOBAL_LIST_EMPTY(mob_config_movespeed_type_lookup)
...
+		
+		if(!.[E.key_third_person])
+			.[E.key_third_person] = list(E)
+		else
+			.[E.key_third_person] |= E
```
Comment (ninjanomnom) [inline code/_globalvars/lists/mobs.dm]:
```suggestion
			.[E.key_third_person] += E
```

File: code/modules/mob/emote.dm [hunk 3]
```diff
@@ -7,12 +7,16 @@
...
 		act = copytext(act, 1, custom_param)
 
-	var/datum/emote/E
-	E = E.emote_list[act]
-	if(!E)
-		to_chat(src, "<span class='notice'>Unusable emote '[act]'. Say *help for a list.</span>")
+
+	var/list/key_emotes = GLOB.emote_list[act]
+
+	if(!key_emotes || !key_emotes.len)
```
Comment (ninjanomnom) [inline code/modules/mob/emote.dm]:
```suggestion
	if(!length(key_emotes))
```

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

File: code/_globalvars/lists/mobs.dm (hunk 1 above)
Comment (ninjanomnom) [reply code/_globalvars/lists/mobs.dm]: If you want to codegolf a bit you could skip the conditional here and just do
`.[E.key] += list(E)`

File: code/_globalvars/lists/mobs.dm (hunk 2 above)
Comment (AnturK) [reply code/_globalvars/lists/mobs.dm]: Technically you can set the key_third_person to the same thing as key here.

**Review by ninjanomnom — APPROVED** [review]

**Pull request merged**